### ISSUE TITLE
Vagrant script seems to not work

### ISSUE THREAD

**Issue by jdg-11** [sireum/case-env#4]
Hi,

The vagrant script seems to not work anymore.  I think, possibly, some of the git repository names may have changed.  When the vagrant script finishes running, the various build tools seem to not be installed.  Perhaps, it's not an issue if these files aren't used anymore?

**Comment by robby-phd** [comment]
It's hard to determine what was going on from your description; a log would be helpful.
The latest CI seL4 build yesterday seems to work (see: https://github.com/sireum/case-env/runs/4590525184?check_suite_focus=true#step:5:11714).

**Comment by jdg-11** [comment]
Wow, thanks for the fast response.  When I dug into it further, I am finding that I can complete the build if I tell it not to build compcert or seL4 tooling.  I'll gather up logs and send them your way.  If there's anything in particular that would help, let me know.

**Comment by robby-phd** [comment]
I made some changes to use a custom Vagrant base/starting box for seL4, which should avoid issues with seL4 dependencies while speeding up the provisioning process.

Please try pulling the changes/rebuild and see if that works better (your log should look similar to https://github.com/sireum/case-env/runs/4609126951?check_suite_focus=true#step:5:1).

**Comment by jdg-11** [comment]
Thanks! The attached logfile from a run this morning.  I just dumped the output to a text file:

[log_vagrant_output-named.txt](https://github.com/sireum/case-env/files/7764940/log_vagrant_output-named.txt)

There were additionally some extra lines printed to the terminal after running for a while:
[ExtraTerminalInfo.txt](https://github.com/sireum/case-env/files/7764953/ExtraTerminalInfo.txt)
 
During the run of the vagrant script, the virtual machine reboots a couple times, and then eventually just asks for a text login.  I let it run for a few (~3.5) hours just to make sure there wasn't something in the background going on and logged in, and it didn't seem like the tooling was there.

If it helps, I spun up a fresh debian VM (both pinned to the case commit and without).  If I set seL4 to not build, and pulled in the seL4 dockerfiles as [seL4-Docker](https://docs.sel4.systems/projects/dockerfiles/), I was able to get a build going.  

A couple other changes that I couldn't find a way around was that after obtaining the case-env repo by git, I had to copy the case-setup.sh to the user's home directory (in the debian VM), and similarly the addons and bin folders had to get copied over to the home directory.  If I didn't do that, I got errors about Sireum not finding things it wanted on the path.  

After those few set up changes, the build ran successfully and I was able to make the Producer-Consumer "hello world" package (http://s21.highassurance.santoslab.org/lectures.html#lectures) in FMIDE and got the HAMR tool to produce a slang project from it, so everything seems working now.

I didn't try to modify the vagrant file though, as I'm not really familiar with that tool.

Let me know if anything else would be helpful.

I'll try the new build you mentioned above; thanks for sharing that!

**Comment by robby-phd** [comment]
Thanks for the logs. Unfortunately, it is caused by a known/open issue in seL4 scripts that might break due Python dependencies: seL4/seL4-CAmkES-L4v-dockerfiles#41

I noticed the same problem you encountered in the following CI build: https://github.com/sireum/case-env/runs/4589684304?check_suite_focus=true#step:5:6329

To workaround it, case-env now uses the latest seL4 scripts/version: https://github.com/sireum/case-env/commit/8fed314e1233f8cb5109d3be39f9143f1aea752e

Moreover, to stop being continually affected, case-env now uses a custom Vagrant base box for seL4 that I previously mentioned.

I'm closing this issue as it has been addressed (and with the originating issue has been identified above).
